utils/include/utils/time.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#ifndef HLASMPLUGIN_UTILS_TIME_H
+#define HLASMPLUGIN_UTILS_TIME_H
+
+#include <compare>
+#include <optional>
+#include <string>
+
+namespace hlasm_plugin::utils {
+
+class timestamp
+{
+    unsigned long long m_year : 18 = 0;
+    unsigned long long m_month : 4 = 0;
+    unsigned long long m_day : 5 = 0;
+    unsigned long long m_hour : 5 = 0;
+    unsigned long long m_minute : 6 = 0;
+    unsigned long long m_second : 6 = 0;
+    unsigned long long m_microsecond : 20 = 0;
+
+#ifndef _MSC_VER
+    unsigned long long as_ull() const noexcept
+    {
+        auto v = (unsigned long long)m_year;
+        v <<= 18;
+        v |= (unsigned long long)m_month;
+        v <<= 4;
+        v |= (unsigned long long)m_day;
+        v <<= 5;
+        v |= (unsigned long long)m_hour;
+        v <<= 5;
+        v |= (unsigned long long)m_minute;
+        v <<= 6;
+        v |= (unsigned long long)m_second;
+        v <<= 6;
+        v |= (unsigned long long)m_microsecond;
```

Comment:
Shouldn't it be this way?

```suggestion
        auto v = (unsigned long long)m_year;
        v <<= 4;
        v |= (unsigned long long)m_month;
        v <<= 5;
        v |= (unsigned long long)m_day;
        v <<= 5;
        v |= (unsigned long long)m_hour;
        v <<= 6;
        v |= (unsigned long long)m_minute;
        v <<= 6;
        v |= (unsigned long long)m_second;
        v <<= 20;
        v |= (unsigned long long)m_microsecond;
```

Reply:
👍 
of course, it should be... 🤦